processors/src/EventProcessor.cxx

```diff
@@ -245,7 +245,11 @@ void EventProcessor::parseVTPData(EVENT::LCGenericObject* vtp_data_lcio)
                     case 2: // HPS Cluster
                         VTPData::hpsCluster  clus;
                         clus.X        = (data      )&0x0003F;
+                        // If the first bit of the index is 1, then it is a negative number
+                        if((clus.X >> 5 & 0x1) == 0x1) clus.X = -((clus.X ^ 0x3F) + 1);
```

Comment:
This was taken from EvioTool, so you might want to correct this there too.

Reply:
Thank you for pointing this out. I think that in EvioTool, the clus.X equivalent is a 6-bit integer, so I think that then works correctly. 
```c++
 struct HPSCluster_t {
    int X:    6;        // bit 0-5
    int Y:    4;        // bit 6-9
    unsigned int E: 13; // bit 10-22
    unsigned int subtype:    4;  // bit 23-26
    unsigned int type:       4;  // bit 27-30
    bool         istype:     1;  // bit 31
    unsigned int T:         10;  // bit 0-9
    unsigned int N:          4;  // bit 10-13   Second word
    unsigned int nothing:   18;  // bit 14-31   Second word not used.
  };
  ```
There is no way to do this outside of a bitwise structure.

Reply:
> This was taken from EvioTool, so you might want to correct this there too.

Thanks so the comments. The VTP cluster bank is actually decoded in VTPCluster.java.
The issue has been taken care in the codes:
        ix = word0 & 0x03F;
        // If the first bit of the index is 1, then it is a negative number 
        if((ix >> 5 & 0x1) == 0x1) ix = -((ix ^ 0x3F) + 1);
        iy = word0 >> 6 & 0x0F;
        // If the first bit of the index is 1, then it is a negative number 
        if((iy >> 3 & 0x1) == 0x1) iy = -((iy ^ 0xF) + 1);